### PR TITLE
doc: Add bitbucket project perms explicit API usage docs

### LINKED ISSUE #36456: Update the Explicit Permissions API documentation to mention Bitbucket Project support

### PR DESCRIPTION
**Pull request by indradhanush**
Fixes #36456.

## Test plan

Doc changes only.

### CODE REVIEW

File: doc/admin/repo/permissions.md [hunk 1]
```diff
@@ -513,6 +513,67 @@ query {
...
+
+To get the state of currently queued or running tasks you can run the following query with a list of project keys (one or more):
+
+```graphql
+query {
+	bitbucketProjectPermissionJobs(projectKeys: ["<project key 1>", "<project key 2>"]) {
+    nodes{
+      InternalJobID,
+      State,
+      FailureMessage,
```
Comment (unknwon) [inline doc/admin/repo/permissions.md:556]: CleanShot 2022-06-20 at 10 02 47@2x
nit: mixed tabs and spaces

File: doc/admin/repo/permissions.md [hunk 2]
```diff
@@ -513,6 +513,67 @@ query {
...
+      Unrestricted,
+    }
+  }
+}
+```
+
+<br />
+
+#### Get information about project permissions job state
+
```
Comment (unknwon) [inline doc/admin/repo/permissions.md:575]: Empty section?

File: doc/admin/repo/permissions.md [hunk 3]
```diff
@@ -513,6 +513,67 @@ query {
...
+Sourcegraph supports setting project wide permissions for Bitbucket code host connections.
+
+#### Setting repository permissions for a project
+
+This API lets site admins set the same permissions for all the users across all the repositories under the project.
+
+First, obtain the project key from the Bitbucket code host.
+
+Next, get the code host ID. Visit the `Manage code hosts` page from the site admin panel in the Sourcegraph instance and click on "Edit" for the code host under which the above project is located. Copy the ID from the URL. For example in the URL https://sourcegraph.example.com/site-admin/external-services/RXh0ZXJuYWxTZXJ2aWNlOjMwNjczNg==, the code host ID is `RXh0ZXJuYWxTZXJ2aWNlOjMwNjczNg==`.
```
Comment (unknwon) [inline doc/admin/repo/permissions.md:526]: Our convention in docs is to use

```
**Site admin > Manage code hosts**
```

Instead of using backticks.

**Review by unknwon — COMMENTED** [review]
Thanks for the docs update!

File: doc/admin/repo/permissions.md [hunk 4]
```diff
@@ -513,6 +513,67 @@ query {
...
+      alwaysNil
+    }
+}
+```
+
+This will return an empty respoinse immediately while also enqueuing a background task to set permissions for all the repositories that belong to the project as identified by the `projectKey` in the API request.
+
+#### Querying project permissions task status
+
+To get the state of currently queued or running tasks you can run the following query with a list of project keys (one or more):
```
Comment (sashaostrikov) [inline doc/admin/repo/permissions.md:548]: let's also add information about `count` and `status` query fields.
here and to the query example

**Review by unknwon — APPROVED** [review]

File: doc/admin/repo/permissions.md [hunk 5]
```diff
@@ -513,6 +513,97 @@ query {
...
+    }
+  }
+}
+```
+
+The API also supports filtering against task `status`, which can be one of the following:
+
+- `queued`
+- `canceled`
+- `failed`
```
Comment (sashaostrikov) [inline doc/admin/repo/permissions.md:576]: status can also be `errored`

File: doc/admin/repo/permissions.md [hunk 6]
```diff
@@ -513,6 +513,97 @@ query {
...
+- `completed`
+
+Additionally, the API supports users to control the number of tasks returned in the output by using the argument `count` upto an upper limit of 500.
```
Comment (sashaostrikov) [inline doc/admin/repo/permissions.md:579]: let's also add information about default count: 100

**Review by sashaostrikov — APPROVED** [review]
LGTM, thanks for writing this!

File: doc/admin/repo/permissions.md (hunk 5 above)
Comment (indradhanush) [inline doc/admin/repo/permissions.md:576]:
```suggestion
- `errored`
- `failed`
```

File: doc/admin/repo/permissions.md (hunk 6 above)
Comment (indradhanush) [inline doc/admin/repo/permissions.md:579]:
```suggestion
Additionally, the API supports users to control the number of tasks returned in the output by using the argument `count` upto an upper limit of 500 with a default value of 100.
```

**Pull request merged**